trace/amqp/integration_test.go

```diff
@@ -0,0 +1,137 @@
+// +build integration
+
+package amqp
+
+import (
+	"context"
+	"testing"
+
+	"github.com/streadway/amqp"
+	"github.com/stretchr/testify/require"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPublisherSuccess(t *testing.T) {
+	ctx := context.Background()
+	pub, err := NewPublisher("amqp://guest:guest@localhost:5672/", "exchangeName")
+	assert.NoError(t, err)
+
+	originalMsg, err := NewJSONMessage(`{"status":"received"}`)
+	assert.NoError(t, err)
+	expectedMsg := `"{\"status\":\"received\"}"`
+
+	msgs := setupRabbitMQConsumer(t)
+	err = pub.Publish(ctx, originalMsg)
+	assert.NoError(t, err)
+
+	received := <-msgs
+	assert.Equal(t, expectedMsg, string(received.Body))
+
+	err = pub.Close(ctx)
+	assert.NoError(t, err)
+}
+
+func TestPublisherFailures(t *testing.T) {
+	type args struct {
+		url          string
+		exchangeName string
+	}
+	tests := map[string]struct {
+		args         args
+		publisherErr string
+	}{
+		"missing url": {
+
+			args{"", ""},
+			"url is required",
```

Comment:
we should use the field also to make it more readable.
```suggestion
 publisherErr:			"url is required",
```